CONTRIBUTING.md

```diff
@@ -0,0 +1,370 @@
+    b = x.createboxfill()
+    b.level_1 = .5
+    b.level_2 = 14.5
+    x.plot(s,b,bg = 1)
+
+    fnm = "test_boxfill_lev1_lev2.png"
+
+    x.png(fnm)
+
+    ret = checkimage.check_result_image(fnm,src,checkimage.defaultThreshold)
+    sys.exit(ret)
+
+#### Running the test suite
+
+The tests can then be run directly inside your build tree by typing::
+
+    ctest
+
+To save time and run tests in Pararallel
+
+    ctest -jN
+
+The tests suite is exhaustive and takes around 20 minutes to run. Often
+it is worth running only a subset of tests first around your changes
+before running the entire suite. This is done using one of the following
+constructs:
+
+    ctest -R test-name
+    ctest -R regex*
+
+#### Running the performance test suite
+
+TODO
+
+### Documenting your code
+
+TODO
+
+Contributing your changes to *UV-CDAT*
+-------------------------------------
+
+### Committing your code
+
+Keep style fixes to a separate commit to make your PR more readable. Once you've made changes, you can see them by typing:
+
+    git status
+
+If you've created a new file, it is not being tracked by git. Add it by
+typing :
+
+    git add path/to/file-to-be-added.py
+
+Doing 'git status' again should give something like :
+
+    # On branch shiny-new-feature
+    #
+    #       modified:   /relative/path/to/file-you-added.py
+    #
+
+Finally, commit your changes to your local repository with an
+explanatory message. An informal commit message format is in effect for
+the project. Please try to adhere to it. Here are some common prefixes
+along with general guidelines for when to use them:
+
+> -   ENH: Enhancement, new functionality
+> -   BUG: Bug fix
+> -   DOC: Additions/updates to documentation
+> -   TST: Additions/updates to tests
+> -   BLD: Updates to the build process/scripts
+> -   PERF: Performance improvement
+> -   CLN: Code cleanup
+
+The following defines how a commit message should be structured. Please
+reference the relevant GitHub issues in your commit message using GH1234
+or \#1234. Either style is fine, but the former is generally preferred:
+
+> -   a subject line with \< 80 chars (50-char subject, 72-char rest).
+> -   One blank line.
+> -   Optionally, a commit message body (72-char).
+
+Now you can commit your changes in your local repository:
+
+    git commit -a
+or
+    git commit -a -m "Message..Here"
+
+If you have multiple commits, it is common to want to combine them into
+one commit, often referred to as "squashing" or "rebasing". This is a
+common request by package maintainers when submitting a Pull Request as
+it maintains a more compact commit history. To rebase your commits:
+
+    git rebase -i HEAD~#
+
+Where \# is the number of commits you want to combine. Then you can pick
+the relevant commit message and discard others.
+
+### Pushing your changes
+
+When you want your changes to appear publicly on your GitHub page, push
+your forked feature branch's commits :
+
+    git push origin shiny-new-feature
```

Comment:
see, it's probably better we mention this in the section where they create the branch, hence my "tracking comment" above.